microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/configordinal/properties/ConfigOrdinalPropertyProvider.java

```diff
@@ -0,0 +1,36 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+* which is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lsp4mp.jdt.internal.configordinal.properties;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.lsp4mp.jdt.core.AbstractStaticPropertiesProvider;
+import org.eclipse.lsp4mp.jdt.core.MicroProfileCorePlugin;
+import org.eclipse.lsp4mp.jdt.core.SearchContext;
+
+/**
+ * Config ordinal property provider.
+ * 
+ * @see https://download.eclipse.org/microprofile/microprofile-config-1.4/microprofile-config-spec.html#_manually_defining_the_ordinal_of_a_built_in_configsource
+ */
+public class ConfigOrdinalPropertyProvider extends AbstractStaticPropertiesProvider {
+
+	public ConfigOrdinalPropertyProvider() {
+		super(MicroProfileCorePlugin.PLUGIN_ID, "/static-properties/mp-configordinal-metadata.json");
+	}
+
+	@Override
+	protected boolean isAdaptedFor(SearchContext context, IProgressMonitor monitor) {
+		return true;

```

Comment:
Would it make sense to check for [`org.eclipse.microprofile.config.Config`](https://github.com/eclipse/microprofile-config/blob/master/api/src/main/java/org/eclipse/microprofile/config/Config.java) as a sanity check ?

Reply:
Yes I think it can be a good idea. Ideally we should check the MP version, but we have no this feature. One idea that I have is to manage a `since` JSON field in the *.medata.json and use it to check if it matches with the Java library version.

But we could do that in an another PR.